### PR TITLE
Tried adding sphere definition into ray code

### PR DESCRIPTION
**Pull request by firebreath1001**
I'm currently working on the code to complete implementation of "Ray Tracing In One Weekend".
- performed cargo format.
- wrote code for sphere if ray intersects.

TODO: FIX necessary, improper image being generated.

### CODE REVIEW

**Comment by de-sh** [comment]
We are supposed to be getting the image below when [using the parameters mentioned in the tutorial](https://raytracing.github.io/books/RayTracingInOneWeekend.html#addingasphere):
desired sphere shaped output
But we end up receiving this:
output


We need to look at the code for `Ray` and `Vec3` to figure out why that is.

File: src/ray.rs [hunk 1]
```diff
@@ -24,8 +22,22 @@ impl Ray {
...
         let t = 0.5 * (unit_dir.y() + 1.0);
         (1.0 - t) * Color::new(1.0, 1.0, 1.0) + t * Color::new(0.5, 0.7, 1.0)
     }
+
+    pub fn hit_sphere(&self, center: Point3, radius: f64) -> bool {
+        let oc = self.orig - center;
+        let (a, b, c) = (
+            self.dir.dot(self.dir),
+            2.0 * oc.dot(self.dir),
+            oc.dot(self.dir) - radius * radius,
```
Comment (de-sh) [inline src/ray.rs:38]: I think you meant
```
oc.dot(oc)
```

File: src/ray.rs [hunk 2]
```diff
@@ -24,8 +22,22 @@ impl Ray {
     }
 
     pub fn color(self) -> Color {
+        if self.hit_sphere(Point3::new(0.0, 0.0, 2.0), 0.01) {
```
Comment (de-sh) [inline src/ray.rs:25]: We should be getting the correct output as shown in the tutorial. This should be
```
if self.hit_sphere(Point3(0.0, 0.0, -1.0), 0.5) { ... }
```

**Review by de-sh — CHANGES_REQUESTED** [review]

File: src/ray.rs (hunk 1 above)
Comment (de-sh) [reply src/ray.rs:38]: This is the cause of all out problems 🤦🏼

Comment (firebreath1001) [reply src/ray.rs:38]: oh :D thanks a lot man!! will patch it up!

Comment (de-sh) [reply src/ray.rs:38]: I patched it up in 161097e, please try implementing more of the code.

**Pull request merged**